inbm/dispatcher-agent/tests/unit/sota/test_os_factory.py

```diff
@@ -33,16 +33,19 @@ def test_raise_error_unsupported_OsFactory(self) -> None:
         self.assertRaises(ValueError, factory.get_os, "MacOS")
 
     def test_create_ubuntu_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_snapshotter('update', '1', False)) \
-            is DebianBasedSnapshot
+        ubuntu_os = SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu')
+        snapshotter = ubuntu_os.create_snapshotter('update', '1', False, True)
+        self.assertIsInstance(snapshotter, DebianBasedSnapshot)
 
     def test_create_yocto_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('YoctoX86_64').create_snapshotter('update', '1', False)) \
-            is YoctoSnapshot
+        yocto_os = SotaOsFactory(self.mock_disp_broker, None, []).get_os('YoctoX86_64')
+        snapshotter = yocto_os.create_snapshotter('update', '1', False, True)
+        self.assertIsInstance(snapshotter, YoctoSnapshot)
 
     def test_create_windows_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Windows').create_snapshotter('update', '1', False)) \
-            is WindowsSnapshot
+        windows_os = SotaOsFactory(self.mock_disp_broker, None, []).get_os('Windows')
+        snapshotter = windows_os.create_snapshotter('update', '1', False, True)
+        self.assertIsInstance(snapshotter, WindowsSnapshot)
 
     def test_create_ubuntu_updater_checker(self) -> None:
         assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_os_updater()) \
```

Comment:
**Suggestion:** Use assertIsInstance instead of assert type for consistency and better readability in all updater checker tests. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def test_create_ubuntu_updater_checker(self) -> None:
        updater = SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_os_updater()
        self.assertIsInstance(updater, DebianBasedUpdater)
```